Sources/ZcashLightClientKit/Block/Utils/InternalSyncProgress.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  File.swift
+//  
+//
+//  Created by Michal Fousek on 23.11.2022.
+//
+
+import Foundation
+
+struct SyncRanges: Equatable {
+    let downloadRange: CompactBlockRange?
+    let scanRange: CompactBlockRange?
+    let enhanceRange: CompactBlockRange?
+    let fetchUTXORange: CompactBlockRange?
+}
+
+protocol InternalSyncProgressStorage {
+    func integer(forKey defaultName: String) -> Int
+    func set(_ value: Int, forKey defaultName: String)
+    @discardableResult func synchronize() -> Bool
+}
+
+extension UserDefaults: InternalSyncProgressStorage { }
+
+actor InternalSyncProgress {
+
+    enum Key: String, CaseIterable {
+        case latestEnhancedHeight
+        case latestUTXOFetchedHeight
+    }
+
+    enum TypeOfWork {
+        /// There is nothing to do. Everything is synced.
+        case none
+        /// These is something to sync.
+        case sync
+        /// Some local height is higher then latest block height from backend API.
+        case higherValue
```

Comment:
I think this is a debug / log worthy situation.

```suggestion
        case wait(latestScannedHeight: BlockHeight, latestBlockHeight: BlockHeight)
```

Reply:
I changed a code a this enum is now gone.